src/app/(app)/payment.tsx

```diff
@@ -29,6 +29,68 @@ export default function Payment() {
           router.back();
         }}
       />
-    </SafeAreaView>
+    </View>
+  );
+};
+
+export default function Payment() {
+  const buyFormRef = useRef<BuyForm>(null);
+  const { mutate } = usePurchaseOrder({});
+
+  const onSubmitHandler: BuyFormSubmitHandler = (data) => {
+    mutate(
+      {
+        order: {
+          credit_card: {
+            card_number: data.cardNumber,
+            exp_month: data.expirationMonth,
+            exp_year: data.expirationYear,
+            cvc: data.cvcCode,
+          },
+          shipping_address: {
+            city: data.city,
+            country: data.country,
+            line_1: data.address1,
+            line_2:
+              typeof data.address2 === 'undefined' || data.address2 === ''
```

Comment:
What about evaluating if `address2` is false like `!address2`? That will take care of those two validations.